source/documentation/deploying-an-app/ingress-with-duplicate-hostnames.html.md.erb

```diff
@@ -0,0 +1,135 @@
+---
+title: Allow Ingress with duplicate hostname across namespaces
+last_reviewed_on: 2025-02-25
+review_in: 6 months
+layout: google-analytics
+---
+
+# <%= current_page.data.title %>
+This guide explains how to deploy applications with duplicate hostnames across different namespaces.
+
+By default, duplicate hostnames across namespaces are not permitted on our platform. However, with specific annotations on your Ingress resources, you can intentionally allow duplicate hostnames for blue–green deployments.
+
+>
+Note:
+Duplicate hostnames are already possible within a single namespace. This guide explains how to extend that capability across different namespaces.
+
+## Required Annotations
+To allow an Ingress to use a hostname that is already in use in another namespace, you **must** add the following three annotations to each Ingress resource (**both** the new and the existing ones):
+
+```
+# Indicates that you intentionally want to allow duplicate hostnames.
+allow-duplicate-host: "true"
+
+# A comma-separated list of namespaces where duplicate hostnames are permitted.
+# IMPORTANT: The value must be identical and in the smae order on both Ingress objects.
+allowed-duplicate-ns: "blue,green"
+
+# Specifies the AWS weight value used for traffic balancing.
+# This value must be non-empty.
+external-dns.alpha.kubernetes.io/aws-weight: "100"
+
+# A unique identifier for the Ingress.
+external-dns.alpha.kubernetes.io/set-identifier: "<ingress-name>-<ns>-${clusterColor}"
+```
+
+## Canary Deployment Example
+
+Because the NGINX Ingress Controller normally does not allow two Ingress objects to define the same host and path combination, you must enable canary ingresses to merge configurations. To do so, add the following canary annotations:
+
+```
+# Marks the Ingress as part of a canary (blue-green) deployment.
+nginx.ingress.kubernetes.io/canary: "true"
+
+# Defines the traffic weight for the canary.
+nginx.ingress.kubernetes.io/canary-weight: "50"
+```
+
+- Baseline Weight Note:
+
+Although both `external-dns.alpha.kubernetes.io/aws-weight` and `nginx.ingress.kubernetes.io/canary-weight` are available, it is often simpler to use a single weight annotation as your baseline for traffic splitting. Consolidating to one value reduces configuration complexity and potential confusion. 
+
+Depending on your environment and the controller’s behavior, you might choose to adjust `nginx.ingress.kubernetes.io/canary-weight` for Canary weighting deployment.
```

Comment:
so should a user always use canary weight over aws-weight? does canary weight adjust the weight value in route 53?